### PR TITLE
Add lab2 - Bubble and Insertion sort

### LINKED ISSUE #595: Lab2 - Bubble sort and Insertion sort
- Implement bubble sort and insertion sort
- Add unit tests

### PR DESCRIPTION
**Pull request by harirakr**
Fixes #595

Review of colleague's PR #587 

#### Changes proposed in this PR:
- Implement Bubble sort and Insertion sort
- Add unit tests

### CODE REVIEW

File: 02_bubble/harirakr/main.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	n := len(s)
+	cs := append([]int{}, s...)
+	for i := n; i > 0; i-- {
+		for j := 0; j < i-1; j++ {
+			if cs[j] > cs[j+1] {
```
Comment (raghunadhpokkalath) [inline 02_bubble/harirakr/main.go:17]: Please see if you can optimize the bubble sort to avoid iteration if the array is already sorted

**Review by raghunadhpokkalath — COMMENTED** [review]
Please try to use  imperative mood in the git commit messages

https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages

**Comment by harirakr** [comment]
> Please try to use imperative mood in the git commit messages
> 
> https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages

File: 02_bubble/harirakr/main.go (hunk 1 above)
Comment (harirakr) [reply 02_bubble/harirakr/main.go:17]: Don.

File: 02_bubble/harirakr/main_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+type Sortfns func([]int) []int
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	assert.Equal(t, "[1 2 3 5]\n", buf.String())
+}
+
+func TestSortFunctions(t *testing.T) {
```
Comment (raghunadhpokkalath) [inline 02_bubble/harirakr/main_test.go:26]: Changes look good to me. One test case can be added to cover negative number sorting also

**Review by juliaogris — APPROVED** [review]
lgtm

**Comment by harirakr** [comment]
Thanks @juliaogris @raghunadhpokkalath

**Pull request merged**